client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -282,10 +282,9 @@ class SimplePaymentsDialog extends Component {
 
 		const { siteId, dispatch, translate } = this.props;
 
-		wpcom
-			.site( siteId )
-			.post( paymentId )
-			.delete()
+		const productAPI = wpcom.site( siteId ).post( paymentId );
+		productAPI.delete()
```

Comment:
Oh, I had a hunch there would be come catch :) I'll need to educate myself about the various REST API flavors and versions.